search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/llm/ModelLocator.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2023 Aryn
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.searchpipelines.questionanswering.generative.llm;
+
+import org.opensearch.client.Client;
+
+/**
+ * Helper class for wiring LLMs based on the model ID.
+ *
+ * TODO Should we extend this use case beyond HttpConnectors/Remote Inference?
+ */
+public class ModelLocator {
+
+    public static Llm getRemoteLlm(String openSearchModelId, Client client) {
```

Comment:
This may warrant a separate discussion.

First, please take a look at this comment I made - https://github.com/opensearch-project/ml-commons/issues/882#issuecomment-1673814168.

Basically, what I had in mind here is not really "remote vs local", but rather "open search remote model vs direct".  In other words, I initially considered having a search pipeline talk directly to OpenAI (and other LLMs) without going through the remote inference framework.  To keep this PR simple, I removed the direct connector implementation and only kept this implementation that relies on a remote model to talk to (external) LLMs.

Reply:
What do you think about the idea of directly consuming the HttpConnector here?  I think it makes sense.  If you know you just need to connect to an external endpoint, why register and deploy a remote model?  Seems like an unnecessary step to me.

Anyway, for this PR, I can change this to just `getLlm`.

Reply:
@austintlee See, the idea of model in ML Commons is to abstract how OpenSearch is interacting with the model. If we start putting direct HTTP connector here, then we are breaking that abstraction. Hence we should only use the Model and related APIs to talk to LLM.